provider/offline.go

```diff
@@ -37,9 +37,36 @@ const (
 	Float64             = "float64"
 	String              = "string"
 	Bool                = "bool"
-	Timestamp           = "time.Time"
+	Timestamp           = "time.Time" //Leaving for backwards compatibility
+	Datetime            = "datetime"
 )
 
+func (vt ValueType) isValid() error {
+	// Check if value type is a valid type
+	switch vt {
+	case NilType, Int, Int32, Int64, Float32, Float64, String, Bool, Timestamp, Datetime:
+		return nil
+	default:
+		return fmt.Errorf("invalid table type: %v", vt)
+	}
+}
+
+func (vt ValueType) doesMatch(value interface{}) bool {
```

Comment:
Forces exact types to be used because of discrepancies in ability to cast with different providers